community/codegen/src/main/java/org/neo4j/codegen/Expression.java

```diff
@@ -153,6 +237,30 @@ public void accept( ExpressionVisitor visitor )
         };
     }
 
+    public static Expression ternaryOnNull( final Expression test, final Expression onTrue, final Expression onFalse )
```

Comment:
Same thing here as for `ifNotStatement`. Instead of having a special method for this ternary, could we not provide more insight into the `test` `Expression` in order to perform this optimisation in the implementation rather than forcing the API user to do it?